### PR TITLE
dconf: add module

### PR DESCRIPTION
**Pull request by rycee**
This module allows unified configuration of dconf settings.

### CODE REVIEW

**Comment by gnidorah** [comment]
@rycee Thank you much for putting time in this! I should really figure how to fix that dbus service issue.

**Comment by rycee** [comment]
Hmm, I certainly have to learn a bit more about Wayland 🙂

But I hope it is possible to manage a Wayland graphical session through systemd. For example, would it be possible to start the compositor as a systemd service? I'm thinking whether it's OK to have it be "wanted by" the graphical-session target.

Do you have a suggestion where I could read up on the way Wayland works?

**Comment by gnidorah** [comment]
@rycee 
> Oh and just to demonstrate the issue with the dbus.socket. On the first run everything goes well, the DBUS_SESSION_BUS_ADDRESS value is set to something like unix:abstract=/tmp/dbus-L2nKDw2gwU,guid=aaa26abd94bb1707c655fa395be08467 which corresponds to the dbus session started by the NixOS xsession script. The requires clause causes dbus.socket unit to start (but the service is not triggered because of the above DBUS_SESSION_BUS_ADDRESS value), dbus.socket does change DBUS_SESSION_BUS_ADDRESS to unix:path=/run/user/1000/bus, though. So next time you start dconf-settings.service it will trigger dbus.service and a second dbus user session is started:

Does it create any issue on your side? Second running dbus server alone doesn't look like a problem.

File: modules/misc/dconf.nix [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    systemd.user.services.dconf-settings =
+      let
+        iniFile = pkgs.writeText "hm-dconf.ini" (toDconfIni cfg.settings);
+      in
+        {
+          Install = {
+            WantedBy = [ "graphical-session-pre.target" ];
+          };
+
+          Unit = {
```
[the remark this answers is not in the document]
Comment (rycee) [reply modules/misc/dconf.nix]: For example, after doing a X11 login through lightdm. The dunst notification daemon is started through systemd and will see the dbus session address set by the dbus session started by the xsession wrapper. I then start a different service that depends on `dbus.socket`, which starts and updates the `DBUS_SESSION_BUS_ADDRESS` value inside systemd. Now I have a third user service that runs, e.g., `notify-send hello`. Dunst doesn't show the notification 🙁

[the remark this answers is not in the document]
Comment (gnidorah) [reply modules/misc/dconf.nix]: @rycee Thanks, I think I should simulate that and think how to workaround that

**Comment by gnidorah** [comment]
Should be replaced by https://github.com/rycee/home-manager/pull/497

**Comment by gnidorah** [comment]
@rycee 

> Please have a look at #436 again, where I incorporated these changes.

> I haven't yet had a chance to try for the NixOS module case.

Works well for me, thank you! Could you please also drop `gtk.gtk3.waylandSupport` option? It was added before dconf solution to workaround error in case of NixOS module. Now `home.activation.gtk3` should be always activated.

**Comment by rycee** [comment]
I've marked `gtk.gtk3.waylandSupport` as removed using `mkRemovedOptionModule`.

**Comment by gnidorah** [comment]
@rycee Thank you much again for working on this issue!

**Comment by eadwu** [comment]
Seems to be caused by this PR?
```
➜ home-manager switch
/nix/store/x0lf49xm1qszchhgwr1438lyk1nwk8d1-home-manager-generation
Starting home manager activation
Activating checkFilesChanged
Activating checkLinkTargets
Activating writeBoundary
Activating installPackages
replacing old 'home-manager-path'
installing 'home-manager-path'
Activating dconfSettings
error: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name ca.desrt.dconf was not provided by any .service files

Usage:
  dconf load DIR

Populate a subpath from stdin

Arguments:
  DIR         A directory path (starting and ending with '/')


There is an unread and relevant news item.
Read it by running the command 'home-manager news'.
```

**Comment by rycee** [comment]
@eadwu Are you on NixOS? If so, please try adding
```
services.dbus.packages = with pkgs; [ gnome3.dconf ]
```
to your system configuration.

**Comment by eadwu** [comment]
Doesn't seem to have changed anything, still getting the error.

**Comment by gnidorah** [comment]
@eadwu reboot?

**Comment by eadwu** [comment]
Rebooting with the configuration fixed it.

**Comment by rycee** [comment]
I created #510 which should make the
```
services.dbus.packages = with pkgs; [ gnome3.dconf ]
```
unnecessary on NixOS but I'm a bit weary of merging that because it could potentially conflict with the system dconf install.

**Pull request merged**